### PR TITLE
feat: add env file support for all sidecar containers

### PR DESCRIPTION
**Pull request by bvtujo**
- chore: add Env File field to manifest and add mft validation
- chore: change how env files are handled in parameters
- chore: change how manifest package handles env files
- chore: update templates to include required policies and Env File keys
- chore: add unit tests for cli/deploy package
- chore: fix cli deploy fields
- test: fix stack integ tests

This PR adds support for env files in all customer-defined sidecars, as well as the managed 
Logging sidecar pattern. 

It does this by creating a new parameter for each sidecar, ${sidecarName}EnvFileARN, and 
the corresponding condition. It uses that condition to modify the execution role policy 
so that the ECS service has access to all env files used in the task. 

The feature uploads each env file only once, allowing customers to specify the same env file
across multiple containers and not waste any space in S3 or bandwidth uploading duplicates.

Addresses #4447.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/manifest/backend_svc.go [hunk 1]
```diff
@@ -126,9 +126,20 @@ func (s *BackendService) BuildArgs(contextDir string) map[string]*DockerBuildArg
...
 }
 
-// EnvFile returns the location of the env file against the ws root directory.
-func (s *BackendService) EnvFile() string {
-	return aws.StringValue(s.TaskConfig.EnvFile)
+// EnvFiles returns the locations of all env files against the ws root directory.
+// This method returns a map[string]string where the keys are container names
+// and the values are either env file paths or empty strings.
+func (s *BackendService) EnvFiles() map[string]string {
+	envFiles := make(map[string]string)
```
Comment (paragbhingre) [inline internal/pkg/manifest/backend_svc.go]: How about this?
plus 2 for main container as well as FirelensContainer. 
```suggestion
	envFiles := make(map[string]string, len(s.Sidecars)+2)
```

File: internal/pkg/manifest/job.go [hunk 2]
```diff
@@ -148,9 +148,20 @@ func (j *ScheduledJob) BuildRequired() (bool, error) {
...
 }
 
-// EnvFile returns the location of the env file against the ws root directory.
-func (j *ScheduledJob) EnvFile() string {
-	return aws.StringValue(j.TaskConfig.EnvFile)
+// EnvFiles returns the locations of all env files against the ws root directory.
+// This method returns a map[string]string where the keys are container names
+// and the values are either env file paths or empty strings.
+func (j *ScheduledJob) EnvFiles() map[string]string {
+	envFiles := make(map[string]string)
```
Comment (paragbhingre) [inline internal/pkg/manifest/job.go]: Samesies as above

**Review by paragbhingre — COMMENTED** [review]
Just a nit, otherwise first pass at it looks good to me.

File: internal/pkg/manifest/backend_svc.go (hunk 1 above)
Comment (bvtujo) [reply internal/pkg/manifest/backend_svc.go]: What's the behavior of a map initialized with a larger length if it's not actually filled out? 

Like, if I'm iterating over it later am I going to have two iterations of my for loop even if there's not actually an env file for the main container or logging sidecar?

Comment (huanjani) [reply internal/pkg/manifest/backend_svc.go]: The length is "approximate" anyway, and more space will be created if necessary. No, it shouldn't iterate over empty spaces. https://go.dev/ref/spec#Making_slices_maps_and_channels

Comment (bvtujo) [reply internal/pkg/manifest/backend_svc.go]: Shipped.

Comment (bvtujo) [reply internal/pkg/manifest/backend_svc.go]: Upon further review, this change represented a magic number with little documentation in the code, and has been removed in favor of readability. The ruling on the field is overturned.

File: internal/pkg/manifest/backend_svc.go [hunk 3]
```diff
@@ -126,9 +126,20 @@ func (s *BackendService) BuildArgs(contextDir string) map[string]*DockerBuildArg
 	return buildArgs
 }
 
-// EnvFile returns the location of the env file against the ws root directory.
-func (s *BackendService) EnvFile() string {
-	return aws.StringValue(s.TaskConfig.EnvFile)
+// EnvFiles returns the locations of all env files against the ws root directory.
+// This method returns a map[string]string where the keys are container names
+// and the values are either env file paths or empty strings.
+func (s *BackendService) EnvFiles() map[string]string {
```
Comment (iamhopaul123) [inline internal/pkg/manifest/backend_svc.go:134]: Looks like the logic for `EnvFiles` funcs are almost identical. Can we possibly reduce the duplication?

File: e2e/sidecars/sidecars_test.go [hunk 4]
```diff
@@ -54,7 +55,9 @@ sidecars:
     image: %s    # Image URL for sidecar container.
     variables:
       NGINX_PORT: %s
+    env_file: ./magic.env
 logging:
+  env_file: ./magic.env
```
Comment (iamhopaul123) [inline e2e/sidecars/sidecars_test.go:60]: Do we need to validate the env files are indeed added to the service?

Comment (bvtujo) [reply e2e/sidecars/sidecars_test.go:60]: yeah, good call

Comment (bvtujo) [reply e2e/sidecars/sidecars_test.go:60]: Done. I settled for writing some logic that checks that there is in fact an env file included with each container. Taken along with the fact that the service won't run if the permissions are wrong or the env file can't be found or uploaded, this should be sufficient.

**Review by huanjani — APPROVED** [review]
:shipit:

**Review by iamhopaul123 — APPROVED** [review]
LGTM. Maybe next time make a smaller PR so that it's safer and easier to review 😅

**Comment by bvtujo** [comment]
Adding a label until I can verify that the reason the sidecars e2e is failing locally at the last step (checking for svc logs) has nothing to do with the changes here.

**Pull request merged**